app/components/UI/Earn/Views/EarnLendingDepositConfirmationView/EarnLendingDepositConfirmationView.test.tsx

```diff
@@ -258,21 +368,108 @@ describe('EarnLendingDepositConfirmationView', () => {
       fireEvent.press(depositButton);
     });
 
-    expect(Engine.controllerMessenger.subscribeOnceIf).toHaveBeenCalledTimes(3);
-
-    expect(mockAddTransaction).toHaveBeenCalledWith(
-      {
-        data: '0x617ba037000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb4800000000000000000000000000000000000000000000000000000000004c4b40000000000000000000000000c4966c0d659d99699bfd7eb54d8fafee40e4a7560000000000000000000000000000000000000000000000000000000000000000',
-        from: '0xC4966c0D659D99699BFD7EB54D8fafEE40e4a756',
-        to: AAVE_V3_ETHEREUM_MAINNET_POOL_CONTRACT_ADDRESS,
-        value: '0',
-      },
-      {
+    expect(
+      Engine.context.EarnController.executeLendingDeposit,
+    ).toHaveBeenCalledWith({
+      amount: '5000000',
+      protocol: 'AAVE v3',
+      underlyingTokenAddress: MOCK_USDC_MAINNET_ASSET.address,
+      gasOptions: {},
+      txOptions: {
         deviceConfirmedOn: 'metamask_mobile',
         networkClientId: 'mainnet',
         origin: 'metamask',
         type: 'lendingDeposit',
       },
+    });
+  });
+
+  it('handles transaction error during approval flow', async () => {
+    const routeParamsWithApproveAction = {
+      ...defaultRouteParams,
+      params: {
+        ...defaultRouteParams.params,
+        action: EARN_LENDING_ACTIONS.ALLOWANCE_INCREASE,
+      },
+    };
+
+    (useRoute as jest.Mock).mockReturnValue(routeParamsWithApproveAction);
+
+    mockAddTransaction.mockRejectedValue(new Error('Transaction failed'));
+
+    const { getByTestId } = renderWithProvider(
+      <EarnLendingDepositConfirmationView />,
+      { state: mockInitialState },
+    );
+
+    const approveButton = getByTestId(
+      CONFIRMATION_FOOTER_BUTTON_TEST_IDS.CONFIRM_BUTTON,
+    );
+
+    await act(async () => {
+      fireEvent.press(approveButton);
+    });
+
+    expect(
+      Engine.context.EarnController.executeLendingTokenApprove,
+    ).toHaveBeenCalled();
+  });
+
+  it('handles transaction error during deposit flow', async () => {
+    mockAddTransaction.mockRejectedValue(new Error('Transaction failed'));
+
+    const { getByTestId } = renderWithProvider(
+      <EarnLendingDepositConfirmationView />,
+      { state: mockInitialState },
+    );
+
+    const depositButton = getByTestId(
+      CONFIRMATION_FOOTER_BUTTON_TEST_IDS.CONFIRM_BUTTON,
     );
+
+    await act(async () => {
+      fireEvent.press(depositButton);
+    });
+
+    expect(
+      Engine.context.EarnController.executeLendingDeposit,
+    ).toHaveBeenCalled();
+  });
+
+  it('displays correct token information in the UI', () => {
```

Comment:
S: only tell if it was expected to be wrong info, otherwise it's implied and if you say "correct" then it means it's not obvious and so you have to define what is correct.
Also, do not specify "in the UI". Where else?
```suggestion
  it('displays token information', () => {
```